File: Core/CommandParser.cs
namespace EpikChatBot.Core;

public readonly record struct ParsedCommand(bool IsCommand, string Command, string Text);

public static class CommandParser
{
  public static ParsedCommand Parse(string? content)
  {
    var text = (content ?? "").Trim();
    if (!text.StartsWith('.') && !text.StartsWith('^')) return new(false, "", text);
    var split = text[1..].Split((char[]?)null, 2, StringSplitOptions.RemoveEmptyEntries);
    return split.Length == 0 ? new(true, "", "") : new(true, split[0].ToLowerInvariant(), split.Length > 1 ? split[1].Trim() : "");
  }
}

File: legacy-node/lib/command-parser.js
"use strict";

function parseCommand(content) {
  const trimmed = typeof content === "string" ? content.trim() : "";
  if (!trimmed.startsWith("^")) return { isCommand: false, command: null, text: trimmed };
  const body = trimmed.slice(1).trim();
  if (!body) return { isCommand: true, command: null, text: "" };
  const [command, ...args] = body.split(/\s+/);
  return { isCommand: true, command: command.toLowerCase(), text: args.join(" ").trim() };
}

module.exports = { parseCommand };
